sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -397,6 +397,10 @@ private[hive] object HiveQl {
   protected def extractDbNameTableName(tableNameParts: Node): (Option[String], String) = {
     val (db, tableName) =
       tableNameParts.getChildren.map { case Token(part, Nil) => cleanIdentifier(part) } match {
+        case Seq(dbNameAndTable) if (dbNameAndTable.contains(".")) => {
+          val tokens = dbNameAndTable.split("\\.")
```

Comment:
Need to check the length of `tokens`  that splited by `\\.` is 2 ? Since we only want `db.table` but not `db.table.xxxx.xxxx`.